lib/newman_conway.rb

```diff
@@ -1,7 +1,26 @@
+# Time complexity: O(n), where n is directly proportional to the size of num
+# Space Complexity: O(n), where n is directly proportional to the size of num
+def newman_conway(num)
+  if num <= 0 
+    raise ArgumentError, "Must pass in integer greater than zero, received: #{num}" 
+  end 
 
+  hash = {1 => 1, 2 => 1}
+  result = []
+  count = 1
 
-# Time complexity: ?
-# Space Complexity: ?
-def newman_conway(num)
-  raise NotImplementedError, "newman_conway isn't implemented"
+  while count <= num 
+    if count == 1 || count == 2 
+      result.push(1)
+    else 
+      last = hash[count - 1] 
+      current = hash[last] + hash[count - last]
+      hash[count] = current 
+      result.push(current) 
```

Comment:
Just curious why a hash, and why name it `hash`, instead of maybe sequence or something to indicate what it stores.